### PR TITLE
Update rke2 calico config for Windows support

### PR DESCRIPTION
**Pull request by manuelbuil**
Gives an alternative to https://github.com/rancher/rke2-charts/pull/111 which will not work due to the Tigera helm chart created for Calico.

Changes:
- All crds (esp. ipamconfigs) are moved into the calico-crd chart
- Required config of ipamconfigs for windows to work gets deployed in the calico chart. My initial tests show that this config does not break linux
- As strictAffinity=true and nodes cannot borrow ips, we increase the amount of usable ips per node from 62 (/24) to 126 (/25) to honor the default k8s config of 110 pods/node.

Signed-off-by: Manuel Buil <mbuil@suse.com>

### CODE REVIEW

**Comment by manuelbuil** [comment]
Tested in linux:

- Ping internode works
- ping intranode works
- network policy works

**Review by rosskirkpat — APPROVED** [review]
If I understand correctly, we would need to monitor upstream changes to the CRDs and merge those in as needed? We would effectively be maintaining a custom Calico chart with this PR. 

I believe this is our only option at this time and have no issue approving this as these changes are required for 2.6.0 functionality because the custom configuration needs deployed before any pods are created.

**Review by brandond — APPROVED** [review]

**Comment by manuelbuil** [comment]
@luthermonson @rosskirkpat Is `allowVXLANPacketsFromWorkloads: true` also a requirement for windows support? Is there a link in the Calico docs that we could write in the comments to point to this required values? That will help in the future to maintain this chart

File: packages/rke2-calico/generated-changes/patch/values.yaml.patch [hunk 1]
```diff
@@ -41,3 +41,10 @@
 +
 +global:
 +  systemDefaultRegistry: ""
++
++ipamConfig:
++  strictAffinity: true
++  autoAllocateBlocks: true
```
Comment (luthermonson) [inline packages/rke2-calico/generated-changes/patch/values.yaml.patch:47]: this is actually not needed, it defaults to true any way so can be dropped

File: packages/rke2-calico/generated-changes/patch/values.yaml.patch [hunk 2]
```diff
@@ -41,3 +41,10 @@
...
++
++felixConfiguration:
++  allowVXLANPacketsFromWorkloads: true
```
Comment (luthermonson) [inline packages/rke2-calico/generated-changes/patch/values.yaml.patch:50]: this defaults to false and is not required to for strict affinity

**Comment by luthermonson** [comment]
@manuelbuil didnt we talk about doing a /24 for block size?

**Comment by luthermonson** [comment]
docs: https://docs.projectcalico.org/getting-started/windows-calico/kubernetes/standard#install-calico-on-linux-control-and-worker-nodes

File: packages/rke2-calico/generated-changes/patch/values.yaml.patch (hunk 2 above)
Comment (manuelbuil) [reply packages/rke2-calico/generated-changes/patch/values.yaml.patch:50]: Ok. Then I suggest leaving the default

File: packages/rke2-calico/generated-changes/patch/values.yaml.patch (hunk 1 above)
Comment (manuelbuil) [reply packages/rke2-calico/generated-changes/patch/values.yaml.patch:47]: We need that parameter when defining an ipamConfig:
```
            required:
            - autoAllocateBlocks
            - strictAffinity
```

**Comment by manuelbuil** [comment]
> @manuelbuil didnt we talk about doing a /24 for block size?

You are right. It was because according to k8s docs 110 is the default pods per node, right? So we would need /25 or lower

**Review by brandond — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]

**Comment by luthermonson** [comment]
https://github.com/rancher/rke2/issues/1289

**Pull request merged**